tutorial/LFRic/background/LFRic_kernel.md

```diff
@@ -0,0 +1,241 @@
+* A field argument (`GH_FIELD`) with the degrees of freedom (DoFs) on `W0`
+  function space that is being updated in this kernel (`GH_INC` access for
+  updating fields on continuous function spaces);
+* A `real`-valued scalar (`GH_REAL`) whose value is read (`GH_READ`) to
+  update the field with.
+
+The `GH_` prefix comes from "GungHo", the name of the dynamical core used in
+the LFRic model. Please refer to the documentation for more information on
+[supported accesses](
+https://psyclone.readthedocs.io/en/stable/dynamo0p3.html#valid-access-modes)
+and [function spaces](
+https://psyclone.readthedocs.io/en/stable/dynamo0p3.html#supported-function-spaces)
+for the LFRic data objects.
+
+The `iterates_over` metadata specifies how LFRic kernels are called
+from the [PSy layer](LFRic_PSy.md). A user-defined kernel in LFRic
+is called from a PSy-layer loop over each cell in the horizontal domain,
+hence the metadata identifier for this way of looping, `CELLS`. This means
+that a kernel operates on a one-cell-wide vertical column of cells.
+
+---
+**NOTE**
+
+* LFRic field data are currently `real`-valued, however there is ongoing
+  work to allow the field data of other intrinsic types, such as `integer`.
+  This will be reflected in the metadata, with `GH_REAL` and `GH_INTEGER`
+  marking the data type and `GH_SCALAR` denoting the scalar arguments. The
+  above `arg_type` notation will change to
+
+  ```fortran
+      type(arg_type), dimension(2) :: meta_args = (/ &
+           arg_type(GH_FIELD,  GH_REAL, GH_INC, W0), &
+           arg_type(GH_SCALAR, GH_REAL, GH_READ)     &
+           /)
+  ```
+
+* The kernel metadata for the iteration spaces are also changing to be
+  clearer about the subset of domain the kernel operates on rather than
+  the PSy-layer looping. In the next PSyclone release `iterates_over = CELLS`
+  will become `operates_on = CELL_COLUMN`.
+---
+
+### `use` statements and encapsulation
+
+Besides inheritance from the abstract `kernel_type`, this kernel requires
+LFRic identifiers for the PSyclone metadata and Fortran kind of the argument
+data. This information is stored in LFRic infrastructure modules.
+
+* `use argument_mod,      only: arg_type, ...` looks for the LFRic
+  identifiers of PSyclone metadata for LFRic objects (e.g. fields);
+* `use fs_continuity_mod, only: ...` looks for the `integer` identifier of
+  the required function space (here `W0`);
+* `use constants_mod,     only: ...` looks for the definitions of Fortran
+  kind (precision) of data in LFRic objects (here `r_def` for `real` and
+  `i_def` for `integer` arguments in LFRic).
+
+The stub example above also illustrates the main "encapsulation rule" in
+LFRic: everything is `private` unless it needs to be accessed outside the
+object. As a rule, object data are `private` as illustrated in the
+`setval_field_w0_kernel_type` definition above. Object procedures can be
+`private` (if used only by the parent object) or `public`ly available
```

Comment:
See previous comment about parent object and visibility.

Reply:
Again, I don't know how the parent came into this. Removed.